python/ray/worker.py

```diff
@@ -442,36 +430,25 @@ def _try_store_and_register(self, object_id, value):
             logger.warning(warning_message)
             self.store_and_register(object_id, value)
 
-    def retrieve_and_deserialize(self, object_ids, timeout, error_timeout=10):
+    def retrieve_and_deserialize(self, object_ids, error_timeout=10):
+        data_metadata_pairs = self.core_worker.get_objects(
+            object_ids, self.current_task_id)
+        assert len(data_metadata_pairs) == len(object_ids)
+
         start_time = time.time()
-        # Only send the warning once.
-        warning_sent = False
         serialization_context = self.get_serialization_context(
             self.current_job_id)
-        while True:
+        results = []
```

Comment:
where does the `while True:` go? I suspect it is needed (to wait for the class definition to arrive).